### ISSUE TITLE
Checkout crashes when Cart Price Rule is removed

### ISSUE THREAD

**Issue by BurningReflex** [magento/magento2#12769]
The checkout "crashes" when the customer has used a coupon code which is being removed while the customer is shopping still.

### Preconditions

1.  Magento 2.2.1

### Steps to reproduce

Ensure you have 2 browsers/tabs open, one for ADMIN, one acting as CUSTOMER.
1. ADMIN -- Create a Cart Price Rule with coupon code with conditions (e.g. 10% discount when product of manufacturer X was added) [Image: Coupon in admin](https://imgur.com/7rwcSvn)
2. CUSTOMER -- Add one or more products to the cart and apply the coupon
3. CUSTOMER -- Ensure the coupon discount is being calculated in the cart [Image: Discount is applied](https://imgur.com/wAUJ2tt)
4. ADMIN -- Remove the Cart Price Rule in the admin
5. CUSTOMER -- Click on "Proceed to Checkout" without changing anything

### Expected result

1. Expected a warning that coupon is no longer valid
2. (Optional) Customer is able to continue the checkout without discount

### Actual result

1. Checkout "crashes", nothing loads and customer can only go back in their browser, refreshing page does not change anything
[Image: Keeps on spinning](https://imgur.com/ZR1Ahyo)

The customer needs to hit "Back" in the browser to return to the cart, at which point the coupon is automatically removed, so the customer can go back to checkout.

**Comment by magento-engcom-team** [comment]
@BurningReflex Thanks for your report. Unfortunately, we were not able to reproduce the issue, by the steps that you provided. We made a [video](https://streamable.com/w8sq6) where we are trying to reproduce the issue. Please watch it and let us know if we missed anything. Also please try to reproduce it with a clean instance of Magento, thank you. https://streamable.com/w8sq6

**Comment by BurningReflex** [comment]
@magento-engcom-team 
Main difference I could find is that you are using 2.2.2, and I was using 2.2.1, so possibly it was fixed in the update. I have yet to update my installation, so I will do this later today and try to reproduce again.

Also; my method was by adding multiple products and I used an attribute for the discount, manufacturer to be precise. But I doubt this would make a difference. If I can't reproduce it on 2.2.2, I'll consider the issue corrected, otherwise I'll reply again.
